src/main/java/com/labsynch/labseer/domain/StandardizationDryRunCompound.java

```diff
@@ -651,6 +653,37 @@ public static TypedQuery<StandardizationDryRunCompound> findStandardizationDryRu
 		return q;
 	}
 
+    public static TypedQuery<StandardizationDryRunCompound> checkForDuplicateStandardizationDryRunCompoundByCdId(Long standardizationDryRunCompoundId, int[] cdIds) {
+		// Given a a standardization dry run compound id ID and list of structure ids
+		// Return a list of StandardizationDryRunCompounds which have theame stereo category and stereo comment as the parent id
+		// exclude the StandardizationDryRunCompound id from the list
+		if(cdIds == null)
+			throw new IllegalArgumentException("The cdIds argument is required");
+		if(standardizationDryRunCompoundId == null)
+			throw new IllegalArgumentException("The standardizationDryRunRowId argument is required");
+
+		StandardizationDryRunCompound standardizationDryRunCompound = findStandardizationDryRunCompound(standardizationDryRunCompoundId);
+		if(standardizationDryRunCompound == null)
+			throw new IllegalArgumentException("The standardizationDryRunRowId argument is invalid - no standardizationDryRunRow with id " + standardizationDryRunCompoundId + " found");
+		
+		EntityManager em = StandardizationDryRunCompound.entityManager();
+		String queryBuilder = "SELECT o FROM StandardizationDryRunCompound AS o WHERE o.CdId IN (:cdIds) AND o.id != :id and o.stereoCategory = :stereoCategory";
+
+		if(standardizationDryRunCompound.getStereoComment() == null) {
+			queryBuilder += " AND o.stereoComment IS NULL";
+		} else {
+			queryBuilder += " AND o.stereoComment = :stereoComment";
```

Comment:
This is not a ignore case match like it is in the code I am replacing. This will be fixed in an upcoming pull request.